### PR TITLE
[TFLite] Change rounding mode of the quantized RESIZE_BILINEAR op for coherence with MultiplyByQuantizedMultiplier

### PR DESCRIPTION
**Pull request by Tessil**
Hello,

This PR changes the rounding mode of the `ResizeBilinearInteger` kernel to be more coherent with the rounding of the `MultiplyByQuantizedMultiplier` method (its single-rounding version #50290).

Thibaut

### CODE REVIEW

**Comment by abattery** [comment]
@jianlijianli @thaink could you review this PR?

**Comment by gbaned** [comment]
@jianlijianli, @thaink Can you please review this PR ? Thanks!

**Comment by jianlijianli** [comment]
Thanks Tessil. The change looks fine. Is it possible to add a test case that behaves differently between the two rounding implementations?

**Comment by Tessil** [comment]
I added two tests with negative values (it seems the resize on negative values wasn't tested). The int16 test would generate a different result with the old rounding implementation.

**Comment by gbaned** [comment]
Hi @jianlijianli Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @jianlijianli Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @jianlijianli Can you please review this PR ? Thank you!

**Review by abattery — CHANGES_REQUESTED** [review]
Could you add e2e test cases in op_tests to validate the behavior with TF?

File: tensorflow/lite/kernels/internal/reference/resize_bilinear.h [hunk 1]
```diff
@@ -212,9 +212,8 @@ inline void ResizeBilinearInteger(
               (input_y - (1 << 10) * y0) * (input_x - (1 << 10) * x0);
           const int64_t output_20 =
               output_20_ll + output_20_lu + output_20_rl + output_20_ru;
-          const int64_t round = (output_20 > 0) ? (1 << 19) : -(1 << 19);
-          const T interpolation =
-              static_cast<T>((output_20 + round) / (1 << 20));
+          const int64_t round = 1 << 19;
```
Comment (abattery) [inline tensorflow/lite/kernels/internal/reference/resize_bilinear.h:215]: How about using the macro to store two implementations, one for two rounding and one for one rounding mode?

**Review by abattery — CHANGES_REQUESTED** [review]

File: tensorflow/lite/kernels/internal/reference/resize_bilinear.h (hunk 1 above)
Comment (Tessil) [reply tensorflow/lite/kernels/internal/reference/resize_bilinear.h:216]: Effectively, now that https://github.com/tensorflow/tensorflow/pull/50290 is merged and the behaviour is behind a TFLITE_SINGLE_ROUNDING flag, I kept the old resize bilinear rounding behaviour and put the new single-rounding one behind TFLITE_SINGLE_ROUNDING.

Thanks

File: tensorflow/lite/kernels/resize_bilinear_test.cc [hunk 2]
```diff
@@ -438,6 +438,25 @@ TEST_P(ResizeBilinearOpTest, HorizontalResizeExtremeValuesInt16) {
               ElementsAreArray(ArrayFloatNear({32765, 32766, 32767})));
 }
 
+TEST_P(ResizeBilinearOpTest, HorizontalResizeExtremeNegativeValuesInt8) {
+  ResizeBilinearOpModel m({TensorType_INT8, {1, 1, 2, 1}}, {1, 3}, GetParam());
+  m.SetInput<int8_t>({-120, -128});
```
Comment (abattery) [inline tensorflow/lite/kernels/resize_bilinear_test.cc:443]: Do we need to have different test comparison per macro?

**Review by abattery — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
Hi @abattery Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @abattery Can you please review this PR ? Thank you!

**Review by alankelly — APPROVED** [review]

**Review by alankelly — APPROVED** [review]

**Review by alankelly — APPROVED** [review]

**Comment by alankelly** [comment]
This was merged on June 10th: https://github.com/tensorflow/tensorflow/commit/c7bc5df59c317689b3b30c6aea2eef7e76131c80

Not sure why it didn't close the issue.

**Pull request merged**